src/Datadog.Trace/Tracer.cs

```diff
@@ -502,19 +503,24 @@ internal async Task WriteDiagnosticLog()
         {
             string agentError = null;
 
-            try
+            // In AAS, the trace agent is deployed alongside the tracer and managed by the tracer
+            // Disable this check as it may hit the trace agent before it is ready to receive requests and give false negatives
+            if (!AzureAppServices.Metadata.IsRelevant)
```

Comment:
I suppose this follows from comment later in the file. If the agent url is explicitly set (to send to another VM perhaps) then we should attempt this check. We should only disable when the agent path was left to the default values.

Reply:
It's not harmful to log this, it just won't have a value.
After named pipes is merged I'll check for the transport and instead log whatever is dictated by that.
The important part here is not failing the initial request and logging API key.

Reply:
I'll follow up on this after the named pipes PR is merged